File: util/sms/src/main/java/top/bettercode/sms/aliyun/AliSmsResponse.java
package top.bettercode.sms.aliyun;

import com.fasterxml.jackson.annotation.JsonProperty;
import top.bettercode.sms.SmsResponse;

public class AliSmsResponse implements SmsResponse {

  @JsonProperty("RequestId")
  private String requestId;

  @JsonProperty("BizId")
  private String bizId;

  @JsonProperty("Code")
  private String code;

  @JsonProperty("Message")
  private String message;

  public String getRequestId() {
    return this.requestId;
  }

  public void setRequestId(String requestId) {
    this.requestId = requestId;
  }

  public String getBizId() {
    return this.bizId;
  }

  public void setBizId(String bizId) {
    this.bizId = bizId;
  }

  public String getCode() {
    return this.code;
  }

  public void setCode(String code) {
    this.code = code;
  }

  @Override
  public String getMessage() {
    return this.message;
  }

  public void setMessage(String message) {
    this.message = message;
  }

  //--------------------------------------------
  @Override
  public boolean isOk() {
    return "OK".equals(code);
  }
}

File: tools/sms/src/main/kotlin/top/bettercode/summer/tools/sms/aliyun/AliSmsResponse.kt
package top.bettercode.summer.tools.sms.aliyun

import com.fasterxml.jackson.annotation.JsonProperty
import top.bettercode.summer.tools.sms.SmsResponse

open class AliSmsResponse : SmsResponse {
    @JsonProperty("RequestId")
    var requestId: String? = null

    @JsonProperty("BizId")
    var bizId: String? = null

    @JsonProperty("Code")
    var code: String? = null

    @JsonProperty("Message")
    override var message: String? = null
    override val isOk: Boolean
        //--------------------------------------------
        get() = "OK" == code
}
